spy.go

```diff
@@ -22,18 +22,23 @@ func ParseCommand(cmd string) (string, []string, error) {
 	return stringResult[0], stringResult[1:], nil
 }
 
-// Execute accepts a command argument and writes the result to a writer
-func Execute(w io.Writer, commandLine string) error {
+// Execute accepts two writers and a shell command
+// It writes to both writers; whilst the first writer logs only
+// the executed command result on the user's terminal
+// the second logs to a specified writer with additional messages
+func Execute(w io.Writer, w2 io.Writer, commandLine string) error {
 	name, args, err := ParseCommand(commandLine)
-	fmt.Fprintln(w, commandLine)
+	m := io.MultiWriter(w, w2)
+
+	w2.Write([]byte("************ START ************\n"))
+	w2.Write([]byte(fmt.Sprintf("COMMAND: %s\n\nOUTPUT:\n", commandLine)))
```

Comment:
Any time you have some string, and you want to send it to a writer, the most convenient way is to call `fmt.Fprintf`—this saves an extra call to `Write` and a `[]byte` conversion. Good to know!